Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -172,65 +178,97 @@ std::map<std::string, std::string> DiscusMultipleScatteringCorrection::validateI
+    }
   } else {
-    if (inputWS->getAxis(0)->unit()->unitID() != "DeltaE")
-      issues["InputWorkspace"] = "Input workspace must have units of DeltaE for inelastic instrument\n";
-    std::set<std::string> axisUnits;
-    axisUnits.insert(SQWS->getAxis(0)->unit()->unitID());
-    axisUnits.insert(SQWS->getAxis(1)->unit()->unitID());
-    if (axisUnits != std::set<std::string>{"DeltaE", "MomentumTransfer"})
-      issues["StructureFactorWorkspace"] +=
-          "S(Q, w) workspace must have units of Energy Transfer and MomentumTransfer\n";
-
-    if (SQWS->getAxis(1)->isSpectra())
-      issues["StructureFactorWorkspace"] += "S(Q, w) must have a numeric spectrum axis\n";
-
-    /* ensure S(Q,w) has some negative values - for a few reasons:
-    1) It's not physical to have a one-sided S(Q,w)
-    2) if S(Q,w) only includes positive w (energy loss) this opens up the possibility of not being able to complete
-       a neutron path through to the detector if the neutron ends up with energy below the min w
-    3) DISCUS took a one sided S(Q,w) and generated values for opposite w using detailed balance so make it clear
-       this algorithm doesn't do that*/
-    if ((axisUnits.find("DeltaE") != axisUnits.end()) && !SQWS->getAxis(1)->isSpectra()) {
-      bool atLeastOnePositive = false;
-      bool xIsW = SQWS->getAxis(0)->unit()->unitID() == "DeltaE";
-      for (size_t iHist = 0; iHist < SQWS->getNumberHistograms(); iHist++) {
-        auto &yValues = SQWS->dataY(iHist);
-        auto wValues = xIsW ? SQWS->dataX(0) : std::vector<double>(yValues.size(), SQWS->getAxis(1)->getValue(iHist));
-        std::vector<std::pair<double, double>> ywVals;
-        std::transform(yValues.begin(), yValues.end(), wValues.begin(), std::back_inserter(ywVals),
-                       [](double y, double w) { return std::make_pair(y, w); });
-        if (std::any_of(ywVals.begin(), ywVals.end(), [](auto yw) { return yw.first > 0. && yw.second < 0.; }))
-          atLeastOnePositive = true;
+    for (auto &SQWS : SQWSs) {
+      if (inputWS->getAxis(0)->unit()->unitID() != "DeltaE")
+        issues["InputWorkspace"] = "Input workspace must have units of DeltaE for inelastic instrument\n";
+      std::set<std::string> axisUnits;
+      axisUnits.insert(SQWS->getAxis(0)->unit()->unitID());
+      axisUnits.insert(SQWS->getAxis(1)->unit()->unitID());
+      if (axisUnits != std::set<std::string>{"DeltaE", "MomentumTransfer"})
+        issues["StructureFactorWorkspace"] +=
+            "S(Q, w) workspace must have units of Energy Transfer and MomentumTransfer\n";
+
+      if (SQWS->getAxis(1)->isSpectra())
+        issues["StructureFactorWorkspace"] += "S(Q, w) must have a numeric spectrum axis\n";
+      std::vector<double> wValues;
+      if (SQWS->getAxis(0)->unit()->unitID() == "DeltaE") {
+        if (!SQWS->isCommonBins())
+          issues["StructureFactorWorkspace"] += "S(Q,w) must have common w values at all Q";
```

Comment:
So the original limitation on needing some negative values wasn't a user requirement. It was more a convenience thing because it made the code simpler to require this (so perhaps laziness on my part). 
A couple of things have changed since I put this limitation in place:

1) The work in this PR meant that the simplifications I was trying to hold on to couldn't be carried forward
2) Spencer has some S(q,w) datasets that don't have any values for w<0 so the algorithm wasn't generating a result for those even though a meaningful result could have been created

So I've removed the validation and it all seems to work fine. I'll mention this in the release note